sklearn/utils/tests/test_array_api.py

```diff
@@ -129,34 +129,35 @@ def test_asarray_with_order_ignored():
 
 
 @pytest.mark.parametrize(
-    "array_namespace, device, dtype_name", yield_namespace_device_dtype_combinations()
+    "array_namespace, device, dtype", yield_namespace_device_dtype_combinations()
 )
 @pytest.mark.parametrize(
-    "sample_weight, normalize, expected",
+    "weights, axis, expected",
     [
-        (None, False, 10.0),
-        (None, True, 2.5),
-        ([0.4, 0.4, 0.5, 0.7], False, 5.5),
-        ([0.4, 0.4, 0.5, 0.7], True, 2.75),
-        ([1, 2, 3, 4], False, 30.0),
-        ([1, 2, 3, 4], True, 3.0),
+        (None, None, 3.5),
+        (None, 0, [2.5, 3.5, 4.5]),
+        (None, 1, [2, 5]),
+        ([0.4, 0.1], 0, [1.6, 2.6, 3.6]),
+        ([0.4, 0.2, 0.2], 1, [1.75, 4.75]),
+        ([1, 2], 0, [3, 4, 5]),
+        ([1, 1, 2], 1, [2.25, 5.25]),
+        ([[1, 2, 3], [1, 2, 3]], 0, [2.5, 3.5, 4.5]),
+        ([[1, 2, 1], [2, 2, 2]], 1, [2, 5]),
     ],
 )
-def test_weighted_sum(
-    array_namespace, device, dtype_name, sample_weight, normalize, expected
-):
+def test_average(array_namespace, device, dtype, weights, axis, expected):
     xp = _array_api_for_tests(array_namespace, device)
-    sample_score = numpy.asarray([1, 2, 3, 4], dtype=dtype_name)
+    sample_score = numpy.asarray([[1, 2, 3], [4, 5, 6]], dtype=dtype)
     sample_score = xp.asarray(sample_score, device=device)
-    if sample_weight is not None:
-        sample_weight = numpy.asarray(sample_weight, dtype=dtype_name)
-        sample_weight = xp.asarray(sample_weight, device=device)
+    if weights is not None:
+        weights = numpy.asarray(weights, dtype=dtype)
+        weights = xp.asarray(weights, device=device)
 
     with config_context(array_api_dispatch=True):
-        result = _weighted_sum(sample_score, sample_weight, normalize)
+        result = _average(sample_score, axis=axis, weights=weights)
 
-    assert isinstance(result, float)
-    assert_allclose(result, expected, atol=_atol_for_type(dtype_name))
+    result = _convert_to_numpy(result, xp)
+    assert_allclose(result, expected, atol=_atol_for_type(dtype))
 
 
```

Comment:
Since we want to maintain our own `_average`, let's extend the test cases to cover the branches reported by codecov.

We can have a dedicated non-regression test function for each exception and expand the existing parameterized function for the `normalize=True/False` argument and boolean weights.